src/main/java/org/javarosa/core/model/SubmissionProfile.java

```diff
@@ -18,71 +20,53 @@
 
 /**
  * A Submission Profile is a class which is responsible for
- * holding and processing the details of how a submission
+ * holding and processing the details of how a <submission/>
  * should be handled.
  *
  * @author ctsims
  */
 public class SubmissionProfile implements Externalizable {
 
-    XPathReference ref;
-    String method;
-    String action;
-    String mediaType;
-    Hashtable<String, String> attributeMap;
+    TreeReference targetref;
+    TreeReference ref;
 
-    public SubmissionProfile() {
+    String resource;
 
-    }
+    public SubmissionProfile() {
 
-    public SubmissionProfile(XPathReference ref, String method, String action, String mediatype) {
-        this(ref, method, action, mediatype, new Hashtable<String, String>());
     }
 
-    public SubmissionProfile(XPathReference ref, String method, String action, String mediatype, Hashtable<String, String> attributeMap) {
-        this.method = method;
+    public SubmissionProfile(String resource, TreeReference targetRef, TreeReference ref) {
         this.ref = ref;
-        this.action = action;
-        this.mediaType = mediatype;
-        this.attributeMap = attributeMap;
-    }
-
-    public XPathReference getRef() {
-        return ref;
-    }
-
-    public String getMethod() {
-        return method;
+        this.targetref = targetRef;
+        this.resource = resource;
     }
 
-    public String getAction() {
-        return action;
+    public TreeReference getTargetRef() {
+        return targetref;
     }
 
-    public String getMediaType() {
-        return mediaType;
+    public TreeReference getRef() {
+        return ref;
     }
 
-    public String getAttribute(String name) {
-        return attributeMap.get(name);
+    public String getResource() {
+        return resource;
     }
 
     @Override
     public void readExternal(DataInputStream in, PrototypeFactory pf) throws IOException, DeserializationException {
-        ref = (XPathReference)ExtUtil.read(in, new ExtWrapTagged(XPathReference.class), pf);
-        method = ExtUtil.readString(in);
-        action = ExtUtil.readString(in);
-        mediaType = ExtUtil.nullIfEmpty(ExtUtil.readString(in));
-        attributeMap = (Hashtable<String, String>)ExtUtil.read(in, new ExtWrapMap(String.class, String.class), pf);
+        this.resource = ExtUtil.readString(in);
```

Comment:
Does changing these `read`- and `writeExternal` signatures not require a migration of some kind?

Reply:
XForm models change too often, so instead of relying on the serialized form, we use the serialized version as a cached object. If deserialization fails, we just reparse the form from the original object, so no migrations should be needed.